common/src/main/java/com/datastax/oss/simulacron/common/stubbing/Prime.java

```diff
@@ -103,10 +103,12 @@ public Prepared toPrepared() {
   public List<Action> getActions(AbstractNode node, Frame frame) {
     if (frame.message instanceof Prepare) {
       if (primedRequest.when instanceof Query) {
-        if (primedRequest.then instanceof SuccessResult
-            || (primedRequest.then instanceof ErrorResult
-                && ((ErrorResult) primedRequest.then).isignoreOnPrepare())) {
+        if (primedRequest.then instanceof SuccessResult) {
           return this.toPreparedAction();
+        } else if (primedRequest.then instanceof ErrorResult) {
+          if (((ErrorResult) primedRequest.then).isignoreOnPrepare()) {
+            return Collections.emptyList();
```

Comment:
Question, in this case we won't return a response for `Prepare` messages at all right?    Shouldn't we still be returning a `PreparedResponse` in this case since we need it to in order for the BoundStatement to fail?

Reply:
In this particular case it will end up [here](https://github.com/datastax/simulacron/blob/46ace5bdcc1b541a358f0dbbd78ec77a8dc6fdcd/native-server/src/main/java/com/datastax/oss/simulacron/server/BoundNode.java#L407) and a response would be sent back.

Reply:
ah ok cool got it, thanks!  Will take another quick look and merge if I don't find any problems.

Reply:
Before merging I'd decide if we want to reuse `ignore_on_prepare` or create a new field `prepared_then`. If we go with the first I'd move `ignore_on_prepare` to `Result` so we can solve https://github.com/datastax/simulacron/pull/24 as well. Sorry, I understand this PR is a bit confusing.

Reply:
👍 I like moving `ignore_on_prepare` to Result.  I'll make that change.

Reply:
I'm almost done with that change,  was more code changes than I expected, but seems to be working good.  Will push this morning and merge.

Reply:
Awesome, thanks for taking a look